schemes/sima_diagnostics/check_energy_gmean_diagnostics.F90

```diff
@@ -0,0 +1,60 @@
+! Diagnostic scheme for check_energy_gmean
+! This replicates the "debug printout" with the global mean total energy in CAM:
+!  nstep, te        0   0.25872620595082335E+10   0.25872620595082335E+10  -0.00000000000000000E+00   0.10126767279027148E+06   0.21940670000000080E+03
+! where the numbers correspond to:
+! - timestep number
+! - global mean input energy using dycore energy formula
+! - global mean output energy at end of physics timestep using dycore energy formula
+! - global mean heating rate for energy conservation
+! - global mean surface pressure
+! - global mean model top pressure
+! These numbers are very useful for matching models bit-for-bit because they include "everything" in the model.
+module check_energy_gmean_diagnostics
+   use ccpp_kinds, only: kind_phys
+
+   implicit none
+   private
+   save
+
+   public :: check_energy_gmean_diagnostics_run  ! main routine
+
+CONTAINS
+
+   !> \section arg_table_check_energy_gmean_diagnostics_run  Argument Table
+   !! \htmlinclude check_energy_gmean_diagnostics_run.html
+   subroutine check_energy_gmean_diagnostics_run( &
+      teinp_glob, &
+      teout_glob, &
+      heat_glob,  &
+      psurf_glob, &
+      ptopb_glob, &
+      errmsg, errflg)
+
+      use spmd_utils,   only: masterproc
+      use cam_logfile,  only: iulog
+      use time_manager, only: get_nstep
+
+      !------------------------------------------------
+      !   Input / output parameters
+      !------------------------------------------------
+      real(kind_phys),    intent(in)  :: teinp_glob    ! global mean energy of input state [J m-2]
+      real(kind_phys),    intent(in)  :: teout_glob    ! global mean energy of output state [J m-2]
+      real(kind_phys),    intent(in)  :: psurf_glob    ! global mean surface pressure [Pa]
+      real(kind_phys),    intent(in)  :: ptopb_glob    ! global mean top boundary pressure [Pa]
+      real(kind_phys),    intent(in)  :: heat_glob     ! global mean heating rate [J kg-1 s-1]
+
+      ! CCPP error handling variables
+      character(len=512), intent(out) :: errmsg
+      integer,            intent(out) :: errflg
+
+      errmsg = ''
+      errflg = 0
+
+      if (masterproc) then
+         write(iulog,'(1x,a9,1x,i8,5(1x,e25.17))') "nstep, te", get_nstep(), teinp_glob, teout_glob, &
+               heat_glob, psurf_glob, ptopb_glob
+      endif
```

Comment:
Thanks for the suggestion! Maybe @PeterHjortLauritzen could chime in on the potential changes. Peter let me know that this is a great set of numbers to make sure everything in the model state matches bit-for-bit, but I agree it could be more useful to format it in a way that makes reading out these numbers easier. I would also advocate to make the same changes in CAM to make comparing logs between CAM and CAM-SIMA easy too.

How much to expand on these logs depends on whether it'll be on or off for default. I could live with it being longer if it was default off - something like
```
energy fixer global means at timestep 5
input energy = 0.25872620595082335E+10   output energy = 0.25872620595082335E+10
heating rate for energy conservation = -0.00000000000000000E+00
surface pressure = 0.10126767279027148E+06 model top = 0.21940670000000080E+03
```

Maybe the surface/top pressures could be removed?

Reply:
It is nice to have all the digits for debugging (quick bit-for-bit testing!). You can probably remove the surface pressure part.

Reply:
I will add an option `&check_energy_gmean_nl`: `print_global_means`. I advocate it to be on by default at least while SIMA is in development or probably controlled by a debug level in the future.

I had hoped to include it in `&check_energy_nl` group but it looks like separate schemes cannot share the same namelist group.

I have tried to keep the output compact yet more informative:
```
 ------------------------
 CAM-SIMA time step advanced (nstep = 82)
 ------------------------
 global mean input energy =   0.25554709900834303E+10 output energy =   0.25554709900834308E+10
 heating rate for energy conservation =   0.26036170396559292E-13
```

Please let me know what you think!

Reply:
looks good to me! Thanks

Reply:
These changes also look good to me, thanks @jimmielin!  I am a little worried that schemes cannot share namelist groups (I thought in theory the CCPP scheme didn't know anything about the actual namelist details), but for now I would say let's just make an issue for it (in ESCOMP/CAM-SIMA) and then deal with it later.  Thanks again!

Reply:
Thanks, added https://github.com/ESCOMP/CAM-SIMA/issues/324

I didn't record the error message but I can fill it in later.